src/rpcmining.cpp

```diff
@@ -61,6 +66,40 @@ UniValue getmininginfo(const UniValue& params, bool fHelp)
         obj.pushKV("kernel-diff-sum",MinerStatus.KernelDiffSum);
     }
 
+    int64_t nMinStakeSplitValue = 0;
+    double dEfficiency = 0;
+    SideStakeAlloc vSideStakeAlloc = {};
+
+    // nMinStakeSplitValue and dEfficiency are out parameters.
+    bool fEnableStakeSplit = GetStakeSplitStatusAndParams(nMinStakeSplitValue, dEfficiency);
+
+    // vSideStakeAlloc is an out parameter.
+    bool fEnableSideStaking = GetSideStakingStatusAndAlloc(vSideStakeAlloc);
+
+    stakesplitting.pushKV("stake-splitting-enabled", fEnableStakeSplit);
+    if (fEnableStakeSplit)
+    {
+        stakesplittingparam.pushKV("min-stake-split-value", nMinStakeSplitValue / COIN);
+        stakesplittingparam.pushKV("efficiency", dEfficiency);
+        stakesplitting.pushKV("stake-splitting-params", stakesplittingparam);
+    }
+    obj.pushKV("stake-splitting", stakesplitting);
+
+    sidestaking.pushKV("side-staking-enabled", fEnableSideStaking);
+    if (fEnableSideStaking)
+    {
+        unsigned int i;
+        for (const auto& alloc : vSideStakeAlloc)
+        {
+            sidestakingvalloc.pushKV("address-" + std::to_string(i), alloc.first);
```

Comment:
Can you use json array for storing list of tuples?

Reply:
Sure. That is a good idea.

Reply:
`  "side-staking": {
    "side-staking-enabled": true,
    "side-staking-allocations": [
      {
        "address": "n34ysBaxXxNjNQnDA94eoqfSkaCsZ54Ytk",
        "allocation-pct": 0.25
      },
      {
        "address": "mhit3mxqGZnhPVZQkNZR6LKAS8SmZeuN4D",
        "allocation-pct": 0.1
      }
    ]
  },
`

Reply:
`  "side-staking": {
    "side-staking-enabled": true,
    "side-staking-allocations": [
      {
        "address": "n34ysBaxXxNjNQnDA94eoqfSkaCsZ54Ytk",
        "allocation-pct": 0.25
      },
      {
        "address": "mhit3mxqGZnhPVZQkNZR6LKAS8SmZeuN4D",
        "allocation-pct": 0.1
      }
    ]
  },
`

Reply:
I hate that markdown changes the formatting even when in code quotes...

I think I need to add an index counter to the array elements. Should I use 0 or 1 based?

Reply:
"side-staking": {
    "side-staking-enabled": true,
    "side-staking-allocations": [
      {
        "id": "0",
        "address": "n34ysBaxXxNjNQnDA94eoqfSkaCsZ54Ytk",
        "allocation-pct": 0.25
      },
      {
        "id": "1",
        "address": "mhit3mxqGZnhPVZQkNZR6LKAS8SmZeuN4D",
        "allocation-pct": 0.1
      }
    ]
  },

Reply:
I do not see why there should be a index.

Reply:
I got rid of it.